File: frontend/src/app/model/TaxAmount.ts
import {BaseEntity} from "./BaseEntity";

export class TaxAmount extends BaseEntity{
  constructor(id?: number, activeSince?: Date, amount?: number);
  constructor(id: number, activeSince: Date, amount: number) {
    super(id);
    this.activeSince = activeSince;
    this.amount = amount;
  }

  public activeSince: Date;
  public amount: number;
}

File: src/main/java/com/apiApp/CateringFacilityAPI/model/jpa/TaxAmount.java
package com.apiApp.CateringFacilityAPI.model.jpa;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Entity
@Table(name = "tax")
public class TaxAmount extends BaseEntity {

    @NotNull
    private LocalDateTime activeSince;

    @NotNull
    private Double amount;

    public LocalDateTime getActiveSince() {
        return activeSince;
    }

    public void setActiveSince(LocalDateTime activeSince) {
        this.activeSince = activeSince;
    }

    public Double getAmount() {
        return amount;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }
}
